app/src/main/java/com/example/cinemapp/ui/main/MovieAdapter.kt

```diff
@@ -0,0 +1,57 @@
+package com.example.cinemapp.ui.main
+
+import android.annotation.SuppressLint
+import android.text.Html
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.bumptech.glide.Glide
+import com.example.cinemapp.BuildConfig
+import com.example.cinemapp.data.Movie
+import com.example.cinemapp.data.MovieCard
+import com.example.cinemapp.databinding.CardMovieBinding
+import java.util.Calendar
+
+class MovieAdapter : RecyclerView.Adapter<MovieAdapter.MovieViewHolder>() {
+
+    private var movies: List<MovieCard> = emptyList()
+
+
+    @SuppressLint("NotifyDataSetChanged")
+    fun setMovies(movies: List<MovieCard>){
+        this.movies = movies
+        notifyDataSetChanged()
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MovieViewHolder {
+        val binding = CardMovieBinding
+            .inflate(LayoutInflater.from(parent.context), parent, false)
+        return MovieViewHolder(binding)
+    }
+
+    override fun onBindViewHolder(holder: MovieViewHolder, position: Int) {
+        holder.bind(movies[position])
+    }
+
+    override fun getItemCount(): Int {
+        return movies.size
+    }
+
+    inner class MovieViewHolder(private val binding: CardMovieBinding) :
+        RecyclerView.ViewHolder(binding.root) {
+        fun bind(movie: MovieCard) {
+            with(binding) {
+                tvTitle.text = movie.title
+                tvTitle.text = Html.fromHtml("<b>${movie.title}</b> (${movie.releaseDate?.let {
```

Comment:
As we agreed, movie card should only contain title and rating, so we can omit this formatting using release date...so just plain simple:

`tvTitle.text = movie.title`

Reply:
I think that, at least for Upcoming movies, the release date is an important piece of information that should be displayed on the card.

Reply:
Yes, but imagine that a Product Owner, who demanded this feature, wanted to show only picture, title and rating. It his/her decision, and it is on you to implement it as best as you can 😅